tests/ert_tests/ert3/stats/test_distributions.py

```diff
@@ -237,3 +274,56 @@ def test_uniform_ppf(lower, upper, q, size, index):
     assert sorted(dist.index) == sorted(ppf_result.index)
     for idx in dist.index:
         assert ppf_result.data[idx] == pytest.approx(expected_value)
+
+
+@pytest.mark.parametrize(
+    ("valueset", "q", "expected_val"),
+    (
+        ([0], 0.0, np.nan),
+        ([0], 0.1, 0),
+        ([0], 0.2, 0),
+        ([0], 0.5, 0),
+        ([1], 0.9, 1),
+        ([42], 0.999, 42),
+        ([0], 1.1, np.nan),
+        ([0], 0.1, 0),
+        ([42, 420, 4200], 0.0, np.nan),
+        ([42, 420, 4200], 0.2, 42),
+        ([42, 420, 4200], 0.332, 42),
+        ([42, 420, 4200], 0.334, 420),
+        ([42, 420, 4200], 0.5, 420),
+        ([42, 420, 4200], 0.665, 420),
+        ([42, 420, 4200], 0.667, 4200),
+        ([42, 420, 4200], 0.9, 4200),
+        ([42, 420, 4200], 0.9, 4200),
+        ([42, 420, 4200], 1.1, np.nan),
+    ),
+)
+def test_discrete_ppf(valueset, q, expected_val):
+
+    for size in (1, 5, 10):
+        dist = ert3.stats.Discrete(valueset, size=size)
+        expected = [expected_val] * size
+        assert len(dist.index) == len(expected)
+
+        ppf_result = dist.ppf(q)
+        assert len(dist.index) == len(ppf_result.data)
+        assert sorted(dist.index) == sorted(ppf_result.index)
```

Comment:
Is the right side of equation correct? ie. `sorted(ppf_result.index)` and not just `ppf_result.index`

Reply:
I am not sure why we only assert the sorted versions of these. The test pass also when removing `sorted()`, but all corresponding tests for other distributions are also relaxed with respect to order.

Reply:
@markusdregi , rumours tell me you have been involved in letting these index lists be sorted prior to asserting?

Reply:
To my best understanding, while analyzing the code, on line 322 neither side requires `sorted` since we haven't specified index upfront. This is not the case for line 338, which might fail if not sorted or making index a set at least.

Reply:
Changed now to not sort.